logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/ClientScope.kt

```diff
@@ -41,7 +46,19 @@ class ClientScope(
             keyPackageLimitsProvider,
             mlsClientProvider
         )
-    val selfClients: SelfClientsUseCase get() = SelfClientsUseCaseImpl(clientRepository)
+
+    // TODO: extract client id provider to it's own class and test it
+    private var _clientId: ClientId? = null
+    private suspend fun clientId(): Either<CoreFailure, ClientId> =
+        if (_clientId != null) Either.Right(_clientId!!) else {
+            clientRepository.currentClientId().onSuccess {
+                _clientId = it
+            }
+        }
+
+    private val clientIdProvider = CurrentClientIdProvider { clientId() }
```

Comment:
`ClientScope` is a child of `UserSessionScope` so can we just pass the `clientIdProvider` from there as a constructor parameter instead of creating another one? 😉

Reply:
Good point 👍